platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -130,15 +130,25 @@ public void writeToFile(final Path newFile) throws IOException {
         if (!file.equals(newFile)) {
             try (final FileChannel fc =
                     FileChannel.open(newFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
-                fileChannel.position(0);
-                MerkleDbFileUtils.completelyTransferFrom(fc, fileChannel, 0, fileChannel.size());
+                // write header
+                writeHeader(fc);
+                // write data
+                writeLongsData(fc);
+                fc.force(true);
             }
         }
     }
 
-    /** No-op as we override writeToFile directly */
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    protected void writeLongsData(final FileChannel fc) {}
+    protected void writeLongsData(final FileChannel fc) throws IOException {
+        final long minValidIndexOffset = minValidIndex.get() * Long.BYTES;
+        fileChannel.position(currentFileHeaderSize + minValidIndexOffset);
```

Comment:
As far as I can read it, this implies the following file structure for `fileChannel`:
* header (including file format version)
* zeros for all longs in [0, minValidIndex) range
* longs for [minValidIndex, size) range

However, `fc` will have different format. It will have just a header and then all longs starting from minValidIndex. Did I get it right?

Reply:
Let me try to clarify it a little bit just to make sure that we're on the same page. In the context of `LongListDisk` implementation `fileChannel` is a source of the data, current file that we're operating on. When we set `minValidIndex`, we don't perform any action one the file channel. The only effect of this update bubbles up here: 
```
final long minValidIndexOffset = minValidIndex.get() * Long.BYTES;
fileChannel.position(currentFileHeaderSize + minValidIndexOffset);
```
That is, before we transfer bytes from the old file to the new file, we move the read position to the min valid index. Then we take into account the fact that we've reduced the file size by `minValidIndexOffset` in the following line:
```
MerkleDbFileUtils.completelyTransferFrom(
                fc, fileChannel, currentFileHeaderSize, fileChannel.size() - minValidIndexOffset);
```